lib/plugin/instance.rb

```diff
@@ -105,8 +106,12 @@ def register_anonymous_cache_key(key, &block)
     Middleware::AnonymousCache.compile_key_builder
   end
 
-  def add_admin_route(label, location)
-    @admin_route = { label: label, location: location }
+  def add_admin_route(label, location, opts = {})
+    @admin_route = { label: label, location: location }.merge(opts.slice(:use_new_show_route))
```

Comment:
I'm open to different naming for `use_new_show_route` ; we need it because we don't want existing routes e.g. `/admin/plugins/calendar` to clash with this new show route when they are using the old `admin.adminPlugins` resource